lang/compiler.go

```diff
@@ -322,13 +318,20 @@ func (p *AstProgram) Start(ctx context.Context, alloc *memory.Allocator) (flux.Q
 		return nil, fmt.Errorf("no interpreter dependencies found")
 	}
 	depsI := deps.(dependencies.Interface)
-	s, err := spec.FromAST(ctx, depsI, astPkg, p.Now)
+	ses, scope, err := p.eval(ctx, depsI)
```

Comment:
`ses` is really hard to get unless finishing reading all the code...

Reply:
I got your point, I could refactor to `sideEffects`.